File: src/main/java/com/system/car/models/Car.java
package com.system.car.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "cars")
@NoArgsConstructor
@Data
public class Car extends Vehicle {

    public Car(Long id, Brand brand, String model, Integer year, List<String> images, String info) {
        super(id, brand, model, year, images, info);
    }

}

File: front/src/app/models/car.ts
import {Brand} from "./brand";

export class Car {
  id: any;
  model: string;
  year: number;
  brand: Brand = new Brand();
  images: string[];

  constructor() {
    this.images = [];
    this.images.push('');
    this.images.push('');
    this.images.push('');
  }
}
